File: REST API/src/main/java/com/edu/models/entities/Account.java
package com.edu.models.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Builder
@Entity
@Table(name = "Account")
@NoArgsConstructor
@AllArgsConstructor
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "AccountID", nullable = false)
    private int id;

    @Column(name = "[Login]", length = 50, unique = true, nullable = false)
    private String login;

    @Column(name = "[Password]", nullable = false)
    private String password;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "UserID", nullable = false)
    @JsonIgnore
    private User user;
}

File: Android Application/app/src/main/java/com/example/mymaptestapp/data/entities/Account.kt
package com.example.mymaptestapp.data.entities

data class Account(
    val id: Int,
    val userId: Int,
    val login: String,
    val password: String
) {
    companion object {
        fun from(map: Map<String, Any>): Account {
            return Account(
                id = map["accountId"] as Int,
                userId = map["userId"] as Int,
                login = map["login"] as String,
                password = map["password"] as String
            )
        }
    }
}
